pyneuroml/plot/PlotMorphologyVispy.py

```diff
@@ -471,6 +486,50 @@ def plot_interactive_3D(
 
     else:
         cell = list(pop_id_vs_cell.values())[0]
+        #Get all segments' distal points
+        segment_points = []
```

Comment:
I wonder if we can move this functionality into the `plot_3D_cell_morphology` function, and add an argument there to activate this transformation (instead of passing `x_angle` and `y_angle` pass: `pcaify=True` (with a better argument name, of course :))? I wonder because this is limited to the case where we're plotting a single cell, so it sort of fits better there. One should also be able to use this when using `plot_3D_cell_morphology` directly?

It'll also mean that the rotation + translation only happens once. At the moment, in the hay model, for example, it's happening multiple times---each time the mesh precision is dropped.

Reply:
Yes this is the goal, this just allowed for visualization of the pca axes for debugging. However, besides not being able to visualize the pca axes after this change, I think there is another issue with the camera, as you noted above. So, the bounding box that's used to center the camera is taken in `plot _interactive_3d` , before the cell is translated to origin or rotated, so the bounding box and therefore the camera, is centered on the original cell's position. Since in this case we're translating the cell to the origin, maybe along with `pcaify=True` , we could set an extra argument in `create_new_vispy_canvas` that sets the `view_center`  of the camera at the origin as well?

Reply:
Yeh. So, if there's only one cell, we don't need to call `create_new_vispy_canvas` in `plot_3D_interactive` at all. I can't remember why I did it there. It doesn't seem necessary. We could conditionalise it there so that it's only called when there's more than one cell to be plotted, otherwise we let `plot_3D_cell_morphology` or `plot_3D_schematic` do it when it finds that `current_canvas` and `current_view` are `None`.

I just remembered that we also have the `plot_3D_schematic` function too, where it'll also be good to apply the PCA method so that the schematic is also upright. So perhaps we need to split out the PCA bits into a new method that takes a cell as argument and returns the "upright" cell? This could be called by both `plot_3D_cell_morphology` and `plot_3D_schematic` before they create the canvas and the meshes etc?

Yes, we could add another optional argument to `create_new_vispy_canvas` called `view_center`, which if provided will be used. If it isn't, we'll calculate the centre from `view_min` and `view_max`?

How does that sound?